### ISSUE TITLE
Define search form as a search landmark

### ISSUE THREAD

**Issue by schalkneethling** [mdn/yari#2613]
Add `role="search"` to our search component to identify it as a search landmark.
https://www.w3.org/TR/wai-aria-1.1/#search

**Comment by peterbe** [comment]
The (soon to come) site-search page has a `<input type="search">` widget too. Do I just need to change that to `<input type="search" role="search">`?

**Comment by schalkneethling** [comment]
> The (soon to come) site-search page has a `<input type="search">` widget too. Do I just need to change that to `<input type="search" role="search">`?

No, this defines the landmark so, you will add that on the container element.

> A landmark region that contains a collection of items and objects that, as a whole, combine to create a search facility.

Thanks, Peter

**Comment by peterbe** [comment]
It might be worth, if I understand it correctly, to wait with this issue until we've landed the new site-search page in Yari. 

Also, do we want to make the search come back to the home page some day? Like it used to be in the olden days.

**Comment by Ryuno-Ki** [comment]
@peterbe In case you want to check for yourself, [WAVE extension](https://wave.webaim.org/extension/) has a tab for landmarks you can use to verify an implementation.

**Comment by peterbe** [comment]
> @peterbe In case you want to check for yourself, [WAVE extension](https://wave.webaim.org/extension/) has a tab for landmarks you can use to verify an implementation.

Where in here does it explain that the search area doesn't have a "landmark"?

<img width="1503" alt="Screen Shot 2021-01-26 at 12 32 48 PM" src="https://user-images.githubusercontent.com/26739/105881568-ac18a800-5fd2-11eb-8e35-874832d8ca9c.png">

Or, perhaps it's more complicated than that.

**Comment by peterbe** [comment]
Without [@schalkneethling 's PR](https://github.com/mdn/yari/pull/2668/files) it's already able to figure out that that area there is the "Search". 
<img width="1502" alt="Screen Shot 2021-01-26 at 12 34 49 PM" src="https://user-images.githubusercontent.com/26739/105881763-e5511800-5fd2-11eb-94f9-494545977eeb.png">

**Comment by Ryuno-Ki** [comment]
You can open the „structure” tab and see which landmarks got detected.

![prod-wave](https://user-images.githubusercontent.com/3097194/105881991-704b0600-6005-11eb-897d-fcf6ac3100ab.png)

for example does not show a landmark (but detects the aria-attribute)

**Comment by schalkneethling** [comment]
Without this role, VoiceOver would list the landmarks as:

```
banner
Main menu navigation
main
complimentary
content information
```

with the `role` it will list:

```
banner
Main menu navigation
search
main
complimentary
content information
```

**Comment by peterbe** [comment]
I thought we finished this? 
What's missing?

**Comment by mattiapontonio** [comment]
> The (soon to come) site-search page has a <input type="search"> widget too. Do I just need to change that to <input type="search" role="search">?

This could be solved with a React component that _always works wherever we put it_.

**Comment by schalkneethling** [comment]
> I thought we finished this?
> What's missing?

I _think_ this is ok now that we have a single search field. I will just double check as I work on the UI. I am fine with closing this one though.
